tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/single/HelmSidecarContainerIT.java

```diff
@@ -0,0 +1,50 @@
+package com.hivemq.helmcharts.single;
+
+import com.hivemq.helmcharts.AbstractHelmChartIT;
+import com.hivemq.helmcharts.util.K8sUtil;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+@Tag("Platform")
+class HelmSidecarContainerIT extends AbstractHelmChartIT {
+
+    @Test
+    @Timeout(value = 5, unit = TimeUnit.MINUTES)
+    void platform_whenSidecarContainer_withConsulTemplateThenLicenseUpdated() throws Exception {
+        K8sUtil.createConfigMap(client,
+                platformNamespace,
+                "consul-template-config-map.yml");
+        installPlatformChartAndWaitToBeRunning("/files/sidecar-containers-test-values.yaml");
+        await().atMost(Duration.ofMinutes(2)).pollInterval(Duration.ofSeconds(5)).untilAsserted(() -> {
+            final var statefulSet =
+                    client.apps().statefulSets().inNamespace(platformNamespace).withName(PLATFORM_RELEASE_NAME).get();
+            assertThat(statefulSet).isNotNull();
+            assertThat(K8sUtil.getHiveMQContainer(statefulSet.getSpec()).getVolumeMounts().stream()) //
+                    .anyMatch(volumeMount -> volumeMount.getName().equals("license-volume") &&
+                            volumeMount.getMountPath().equals("/opt/hivemq/license"));
+            assertThat(K8sUtil.getContainer(statefulSet.getSpec(), "consul-template").getVolumeMounts().stream()) //
+                    .anyMatch(volumeMount -> volumeMount.getName().equals("license-volume") &&
+                            volumeMount.getMountPath().equals("/opt/hivemq/license"));
+
+            final var volumes = statefulSet.getSpec().getTemplate().getSpec().getVolumes();
+            assertThat(volumes).isNotNull();
+            assertThat(volumes.stream()) //
+                    .anyMatch(volume -> volume.getName().equals("license-volume") &&
+                            volume.getEmptyDir() != null);
+        });
+        waitForPlatformLog(".*License file license.lic is corrupt.");
+    }
+
+    @Override
+    protected boolean deletePlatformNamespace() {
+        // Cannot delete namespace as Sidecar containers will remain in the pod even if the main container is stopped
```

Comment:
Hmm, doesn't the whole Pod get terminated and sends a SIGTERM or SIGKILL to all containers? I guess your sample sidecars are just not reacting to this, which would leave them running for the whole remaining testsuite? Can we try to fix this?

In the Platform Operator Init App tests I'm using this bash script as a fake broker `run.sh`:
```shell
#!/usr/bin/env bash

terminate() {
  echo "Termination signal received. Exiting gracefully..."
  exit 0
}

# trap the termination signal
trap terminate SIGTERM

# loop to keep the process alive
while true; do
  sleep 1
done
```
This should be stoppable, but otherwise runs in an infinite loop.